src/taco-utils/projectHelper.ts

```diff
@@ -0,0 +1,362 @@
+/// <reference path="../typings/tacoProjectInfo.d.ts" />
+
+import child_process = require("child_process");
+import fs = require ("fs");
+import Q = require ("q");
+import path = require ("path");
+import wrench = require ("wrench");
+
+import cordovaHelper = require ("./cordovaHelper");
+import errorHelper = require ("./tacoErrorHelper");
+import jsonSerializer = require ("./jsonSerializer");
+import logger = require ("./logger");
+import tacoErrorCodes = require ("./tacoErrorCodes");
+import telemetryHelper = require ("./telemetryHelper");
+
+import CordovaHelper = cordovaHelper.CordovaHelper;
+import ICommandTelemetryProperties = telemetryHelper.ICommandTelemetryProperties;
+import JsonSerializer = jsonSerializer.JsonSerializer;
+import Logger = logger.Logger;
+import TacoErrorCodes = tacoErrorCodes.TacoErrorCode;
+import TelemetryHelper = telemetryHelper.TelemetryHelper;
+
+module TacoUtility {
+    interface IPluginFetchInfo {
+        [plugin: string]: {
+            source: {
+                type: string;
+                id: string;
+            };
+            "is_top_level": boolean;
+        };
+    }
+    /**
+     *  A helper class with methods to query the project root, project info like CLI/kit version etc.
+     */
+    export class ProjectHelper {
+        private static TACO_JSON_FILENAME: string = "taco.json";
+        private static CONFIG_XML_FILENAME: string = "config.xml";
+        private static PROJECTS_SCRIPTS_DIR: string = "scripts";
+
+        private static cachedProjectInfo: Q.Promise<IProjectInfo> = null;
+        private static cachedProjectFilePath: string = null;
+
+        /**
+         *  Helper to create the taco.json file in the project root {projectPath}. Invoked by
+         *  the create command handler after the actual project creation  
+         */
+        public static editTacoJsonFile(editParams: ITacoJsonEditParams, cordovaCliVersion: string): Q.Promise<any> {
+            ProjectHelper.cachedProjectInfo = null;
+            ProjectHelper.cachedProjectFilePath = null;
+            var deferred: Q.Deferred<any> = Q.defer<any>();
+            var tacoJsonPath: string = path.resolve(editParams.projectPath, ProjectHelper.TACO_JSON_FILENAME);
+            var tacoJson: ITacoJsonMetadata = fs.existsSync(tacoJsonPath) ? require(tacoJsonPath) : {};
+
+            if (editParams.isKitProject) {
+                tacoJson.kit = editParams.version;
+                tacoJson["cordova-cli"] = cordovaCliVersion;
+                return ProjectHelper.createJsonFileWithContents(tacoJsonPath, tacoJson);
+            } else {
+                if (tacoJson.kit) {
+                    delete tacoJson.kit;
+                }
+
+                if (!editParams.version) {
+                    deferred.reject(errorHelper.get(TacoErrorCodes.CommandCreateTacoJsonFileCreationError));
+                    return deferred.promise;
+                }
+
+                tacoJson["cordova-cli"] = editParams.version;
```

Comment:
this requires some cleanup as part of another change. We really don't need tacoEditParams and we can just use tacoJsonMetadata here. version used to represent kit_version || cli_version but with recent changes (which always add cordova-cli field to taco.json) it would make more sense to rename the field. I will let @meenakb  drive that cleanup :)